src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -88,7 +89,8 @@ public CommandResult execute(Model model) throws CommandException {
         Task editedTask = createEditedTask(taskToEdit, editTaskDescriptor, model);
 
         //if the updated date is not same as original and if date is not today onwards
-        if (!taskToEdit.getDate().equals(editedTask.getDate()) && editedTask.getDate().isBefore(LocalDate.now(clock))) {
+        if (!taskToEdit.getDate().equals(editedTask.getDate())
+                && editedTask.getDate().isBefore(GlobalClock.dateToday())) {
```

Comment:
Pls also remove the clock variables that it is currently using and remove the overload the constructor created for the clock

Reply:
Fixed